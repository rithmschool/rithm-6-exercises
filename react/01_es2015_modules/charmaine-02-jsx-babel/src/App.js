import React, { Component } from 'react';
import { FirstComponent } from "./FirstComponent";
import { SecondComponent}  from "./SecondComponent";
import { NamedComponent } from "./NamedComponent";
import logo from './logo.svg';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">My First React App</h1>
        </header>
        <FirstComponent/>
        <SecondComponent/>
        <NamedComponent/>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
