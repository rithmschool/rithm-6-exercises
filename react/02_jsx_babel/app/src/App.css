.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.tweet-box {
  border: solid 1px grey;
  border-radius: 10px;
  min-width: 20%;
  max-width: 25%;
  margin: 5px auto;
  text-align: left;
}

.tweet-username-date {
  display: flex;
  justify-content: space-between;
  margin: 15px 10px 5px 5px;
}

.tweet-username {
  text-decoration: underline;
  color: blue;
}

.tweet-date {
  color: rgba(0,0,0, .9)
}

.tweet-user-name {
  margin: 10px 5px 10px 5px;
  color: darkgray;
}

.tweet-message {
  margin: 10px 7.5px 15px 7.5px;
}

.hobbies {
  list-style-type: none;
}
