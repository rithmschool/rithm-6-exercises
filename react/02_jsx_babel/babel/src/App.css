.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.Tweet-container {
  background-color: lightcyan;
}

.Tweet-name-date {
  color: lightblue;
}

ul {
  margin-left: 35%;
  text-align: left;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
