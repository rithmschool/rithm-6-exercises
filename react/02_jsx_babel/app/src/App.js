import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import FC from './FirstComponent';
import SC from './SecondComponent';
import NC from './NamedComponent';
import Tweet from './Tweet'
import Person from './Person'

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <h1>
          PART 1
        </h1>
        <FC/>
        <SC/>
        <NC name="Jim"/>
        <hr/>
        
        <h1>
          PART 2
        </h1>
        <div className="tweets">
          <Tweet username="sir-loves-a-lot" date="3/20" name="Don John" message="This is my first tweet, it is so exciting"/>
          <Tweet username="kicker" date="3/21" name="Mary Ann" message="I just got on Tweeter, but I already hate this Don John guy!"/>
          <Tweet username="blond-hurricane" date="3/22" name="Lee Onora" message="Common @kicker, chill! He's just a troll from facebook"/>
        </div>
        <hr/>
        
        <h1>
          PART 3
        </h1>
        <Person name="Leona" age="26" hobbies={["stuff", "more stuff"]}/>
        <Person name="Randolphus" age="20" hobbies={["hobby1", "hobby2"]}/>
        <Person name="Ellixirina" age="18" hobbies={["pokemon go", "walking"]}/>
        <hr/>

      </div>
    );
  }
}

export default App;
