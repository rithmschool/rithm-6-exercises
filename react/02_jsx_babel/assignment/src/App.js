import React, { Component } from "react";
import First from "./FirstComponent";
import Second from "./SecondComponent";
import Named from "./NamedComponent";
import Tweet from "./Tweet";
import Person from "./Person";
import logo from "./logo.svg";
import "./App.css";

class App extends Component {
  render() {
    return (
      <div>
        <div>
          Part 1
          <First />
          <Second />
          <Named name="niki" />
        </div>
        <hr />
        <div>
          Part 2
          <Tweet
            username="ILoveSandWhiches"
            user="niki"
            date="Tuesday"
            message="Howdy"
          />
          <Tweet
            username="ILoveSandWhiches"
            user="Cyrus"
            date="Tuesday"
            message="I Like People"
          />
          <Tweet
            username="ILoveSandWhiches"
            user="Xeres"
            date="Tuesday"
            message="I Don't"
          />
        </div>
        <hr />
        <div>
          Part 3
          <Person
            name="Niki"
            age={24}
            hobbies={["coding", "codin", "coughding"]}
          />
        </div>
      </div>
    );
  }
}

export default App;
