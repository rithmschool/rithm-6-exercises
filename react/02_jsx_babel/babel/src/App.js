import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import FirstComp from './FirstComponent';
import SecondComp from './SecondComponent';
import NamedComp from './NamedComponent';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          <FirstComp />
          <SecondComp />
          <NamedComp name="Karl" />
        </p>
      </div>
    );
  }
}

export default App;
