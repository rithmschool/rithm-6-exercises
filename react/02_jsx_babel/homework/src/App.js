import React, { Component } from "react";
import logo from "./logo.svg";
import "./App.css";
import FirstComponent from "./FirstComponent.js";
import SecondComponent from "./SecondComponent.js";
import NameComponent from "./NameComponent";
import Tweet from "./tweet";
import MultiTweet from "./MultiTweet";
import Person from "./Person";
import MultiPerson from "./MultiPerson"

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <FirstComponent />
        <SecondComponent />
        <Tweet
          username="@Max"
          name="Max"
          date="February 29 1992"
          message="HELLLOOOOO"
        />
        <MultiTweet />
        <NameComponent name="Max" />
        <Person
          age="22"
          name="MAXXXXXdafdfafdadfas"
          hobbies={[
            "code",
            "ski",
            "resight Shakespare"
          ]}
        />
        <MultiPerson />
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
