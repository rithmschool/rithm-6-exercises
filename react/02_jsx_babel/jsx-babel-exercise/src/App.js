import React, { Component } from "react";
import logo from "./logo.svg";
import "./App.css";
import FirstComponent from "./FirstComponent.js";
import SecondComponent from "./SecondComponent.js";
import ThirdComponent from "./ThirdComponent.js";
import Tweet from "./Tweet.js";
import Person from "./Person.js";

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        {/* <div>
          <FirstComponent />
          <SecondComponent />
          <ThirdComponent name="Miranda" />
        </div> */}
        <div>
          <Tweet
            username="PaulyG"
            name="Paula"
            date="4/20/18"
            message="React is okay... I guess"
          />
          <Tweet
            username="charlemagne"
            name="Charmaine"
            date="4/01/18"
            message="React is almost as cool as Rithm"
          />
          <Tweet
            username="mirandaHow"
            name="Miranda"
            date="4/11/18"
            message="React is hella cool"
          />
        </div>

        <Person
          name="Paula"
          age="26"
          hobbies={["PokemonGo", "Cuddling Pigs", "Backgammon"]}
        />
        <Person
          name="Alex"
          age="21"
          hobbies={["Crushing Babies", "React", "Lumbering"]}
        />
        <Person
          name="Charmaine"
          age="29"
          hobbies={["Sippin Rose", "Instagramming", "Beating up old people"]}
        />
      </div>
    );
  }
}

export default App;
