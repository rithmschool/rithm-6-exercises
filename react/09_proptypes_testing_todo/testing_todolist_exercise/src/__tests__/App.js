import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../App';


describe('<App />', () => {
  let wrapper;
  it('renders', () => {
    wrapper = shallow(<App />);
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
