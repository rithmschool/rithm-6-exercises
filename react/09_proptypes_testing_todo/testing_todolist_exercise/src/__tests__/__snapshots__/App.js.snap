// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      To Do List
    </h1>
  </header>
  <TodoList />
</div>
`;
